pandas/core/frame.py

```diff
@@ -4411,22 +4411,27 @@ def corr(self, method='pearson', min_periods=1):
         else:
             if min_periods is None:
                 min_periods = 1
-            mat = mat.T
+            mat = com._ensure_float64(mat).T
             corrf = nanops.get_corr_func(method)
             K = len(cols)
             correl = np.empty((K, K), dtype=float)
             mask = np.isfinite(mat)
             for i, ac in enumerate(mat):
                 for j, bc in enumerate(mat):
-                    valid = mask[i] & mask[j]
-                    if valid.sum() < min_periods:
-                        c = NA
-                    elif not valid.all():
-                        c = corrf(ac[valid], bc[valid])
+                    if i > j:
+                        continue
+                    elif i == j:
+                        correl[i, i] = 1.
```

Comment:
this is not correct if all of the values are ``NaN``, then the result is ``Nan``, so you can check that case or let this fall thru. Pls test for this (if its not done already)